src/deluge/dsp/delay/delay_buffer.cpp

```diff
@@ -133,8 +133,10 @@ void DelayBuffer::setupForRender(int32_t userDelayRate, DelayBufferSetup* setup)
 
 	if (isResampling) {
 
-		setup->actualSpinRate = ((uint64_t)userDelayRate << 24) / nativeRate; // 1 is represented as 16777216
-		setup->divideByRate = 0xFFFFFFFF / (setup->actualSpinRate >> 8);      // 1 is represented as 65536
+		setup->actualSpinRate =
+		    (uint64_t)((float)(userDelayRate << 24) / (float)nativeRate); // 1 is represented as 16777216
+		setup->divideByRate =
+		    (uint32_t)((float)0xFFFFFFFF / (float)(setup->actualSpinRate >> 8)); // 1 is represented as 65536
```

Comment:
this could use an intermediate variable to avoid converting float to int to float